misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/ObjectMock.java

```diff
@@ -324,6 +326,12 @@ private static boolean validateReturnValue(
                     List<Type> memberTypes =
                             ((UnionType) attachedFunction.getType().getReturnParameterType()).getMemberTypes();
                     for (Type memberType : memberTypes) {
+                        if (memberType.getTag() == TypeTags.PARAMETERIZED_TYPE_TAG) {
+                            Type bObjectType = ((ParameterizedType) memberType).getParamValueType();
+                            if (TypeChecker.checkIsType(returnVal, bObjectType)) {
+                                return true;
+                            }
+                        }
                         if (TypeChecker.checkIsType(returnVal, memberType)) {
```

Comment:
If the previous conditions result in false, will this still try to compare to the null value? 
Should we we add an if-else within the loop to handle the `ParameterizedTypeTag` case and otherwise?

Reply:
Implemented the if-else as suggested .